adding-behavior/changing-the-dom.md

```diff
@@ -0,0 +1,255 @@
+
+![The Crab Fan Site with the heading "Qualities and Traits of a Crab"](../assets/adding-behavior_changing-the-dom_text-content.png)
+
+### Example: Changing Styles
+
+How do we change the appearance of an element using JavaScript? We'll mix in our CSS skills, and add a CSS class to our target element!
+
+Keira's [`index.html` page](https://replit.com/@adacore/Changing-the-DOM-Demo#index.html) contains a very important fact in one element. This element has an `id` of `appearance__fact--important`.
+
+Keira wants to highlight this important fact and give it a nice yellow background color, and make the text bolder.
+
+She can create a new CSS class, `highlight`:
+
+```css
+.highlight {
+  background-color: palegoldenrod;
+  font-weight: bolder;
+}
+```
+
+Then, she can select the important fact element and add this new class to the element:
+
+```js
+const importantFact = document.getElementById("appearance__fact--important");
+
+importantFact.className = `${importantFact.className} highlight`;
+```
+
+`className` is an attribute, not a method.
+
+- To read the list of classes, we use the dot-operator
+- To set the list of classes, we use the assignment operator
+
+Note that we assign `importantFact.className` to a string that interpolates the existing `importantFact.className` list! Since we want to _add_ the class `highlight`, and not _replace_ all classes in the list, we're including the current list of the element's classes.
+
+![The Crab Fan Site with one important fact highlighted with a pale yellow background and bolder text](../assets/adding-behavior_changing-the-dom_appearance.png)
+
+## Creating New Elements
+
+Keira's really grateful for whoever wrote this initial Crab Fan Site `index.html`, but there just isn't enough crab facts. Keira needs to add another fact to the list.
+
+In order to add new content to our page using JavaScript, we again approach this problem by thinking about the DOM.
+
+We'll combine these two methods for our strategy:
+
+1. `document` has a method named `createElement`, which creates a new HTML element.
+1. Every HTML element has a method named `appendChild`. This method adds a new child node to the element as the last child node.
+
+After Keira looks up online how to use the methods `createElement` and `appendChild`, she can write [this code](https://replit.com/@adacore/Changing-the-DOM-Demo#scripts/index.js):
+
+```js
+const newAppearanceFact = document.createElement("li");
+
+newAppearanceFact.textContent = "covered in sand (when on a sandy beach)";
+
+const appearanceList = document.getElementById("appearance__list");
+
+appearanceList.appendChild(newAppearanceFact);
+```
+
+In [this example](https://replit.com/@adacore/Changing-the-DOM-Demo#scripts/index.js), after reloading the page, we should see a new, sixth crab fact.
+
+![The Crab Fan Site with a new fact added to the list, "covered in sand (when on a sandy beach)"](../assets/adding-behavior_changing-the-dom_adding-content.png)
```

Comment:
adds caption
```suggestion
![The Crab Fan Site with a new fact added to the list, "covered in sand (when on a sandy beach)"](../assets/adding-behavior_changing-the-dom_adding-content.png)  
_Fig. Keira's crab fan site laying down the facts thanks to JavaScript!_
```

Reply:
It would be nice if this image were shorter vertically. The bottom third or so doesn't convey any useful information.